projects/media-viewer/src/lib/store/effects/document.effects.ts

```diff
@@ -0,0 +1,31 @@
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Injectable } from '@angular/core';
+import { HttpResponse } from '@angular/common/http';
+import { catchError, exhaustMap, map } from 'rxjs/operators';
+import { DocumentConversionApiService } from '../../viewers/convertible-content-viewer/document-conversion-api.service';
+import { of } from 'rxjs';
+import * as documentActions from '../actions/document.action';
+
+@Injectable()
+export class DocumentEffects {
+  constructor(
+    private actions$: Actions,
+    private documentConversionService: DocumentConversionApiService,
+  ) { }
+
+  @Effect()
+  covert$ = this.actions$.pipe(
+    ofType(documentActions.CONVERT),
+    map((action: documentActions.Convert) => action.payload),
+    exhaustMap((docConversionPayload) => {
+      return this.documentConversionService.convert(docConversionPayload).pipe(
+        map((result: HttpResponse<Blob>) => {
+          const url = URL.createObjectURL(result.body);
+          return new documentActions.ConvertSuccess(url);
+        }),
+        catchError(error => {
+          return of(new documentActions.ConvertFail(error));
```

Comment:
do we need the of() here, does it work without it?

Reply:
Yes, it doesn't compile without it